semgrep-core/src/parsing/tree_sitter/Parse_swift_tree_sitter.ml

```diff
@@ -3030,28 +3037,22 @@ let map_global_declaration (env : env) (x : CST.global_declaration) =
   | `Prec_group_decl x -> map_precedence_group_declaration env x
   | `Asso_decl x -> map_associatedtype_declaration env x
 
-let map_top_level_statement (env : env) (x : CST.top_level_statement) =
+let map_top_level_statement (env : env) (x : CST.top_level_statement)
+    (semi : CST.semi) =
   match x with
-  | `Exp x -> map_expression env x
+  | `Exp x ->
+      let expr = map_expression env x in
+      let semi = token env semi in
+      G.ExprStmt (expr, semi) |> G.s
   | `Global_decl x -> map_global_declaration env x
   | `Labe_stmt x -> map_labeled_statement env x
-  | `Throw_stmt x -> map_throw_statement env x
+  | `Throw_stmt x -> map_throw_statement env x |> todo env
 
-let map_source_file (env : env) ((v1, v2) : CST.source_file) =
-  let v1 =
-    match v1 with
-    | Some x -> map_shebang_line env x
-    | None -> todo env ()
-  in
-  let v2 =
-    List.map
-      (fun (v1, v2) ->
-        let v1 = map_top_level_statement env v1 in
-        let v2 = (* semi *) token env v2 in
-        todo env (v1, v2))
-      v2
+let map_source_file (env : env) ((_shebang, stmts) : CST.source_file) : G.any =
+  let stmts =
+    List.map (fun (stmt, semi) -> map_top_level_statement env stmt semi) stmts
```

Comment:
Just FYI I think we've run into memory issues with `List.map`; you may want to `List.rev List.rev_map`

Reply:
If this is about stack overflows, we have `Common.map`.